src/Addon/PluginUpgrade.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace GiveAddon\Addon\Helpers;
+
+/**
+ * Class PluginUpgrade
+ * @package GiveAddon\Infrastructure
+ *
+ * @unreleased
+ */
+class PluginUpgrade {
+    /**
+     * @unreleased
+     */
+    public function storePluginUpgradeVersion() {
+        $pluginVersion = preg_replace( '/[^0-9.].*/', '', get_option( 'give_addon_version' ) );
+
+        // Is Fresh install?
+        if ( ! $pluginVersion ) {
+            $pluginVersion = '1.0.0';
```

Comment:
This seems too tightly coupled to a specific version, for which there are other possible starting points.